5a/Client/src/Program.fs

```diff
@@ -82,17 +85,20 @@ let main _args =
     client.Connect
         (fun error ->
             task {
-                if not (error = null) then
-                    printfn $"Error connecting to cluster: ${error}"
-                    return false
-                else
-                    printfn "Connected to cluster"
-                    return true
+                Async.Sleep(TimeSpan.FromSeconds(1)) |> Async.RunSynchronously
+                attempts <- attempts + 1
+                printfn $"Failed to connect to cluster on attempt {attempts} of {maxAttempts}"
+
+                return if attempts > maxAttempts then false
+                    else not(error = null)
             })
     |> Async.AwaitTask
     |> Async.RunSynchronously
 
-    let grain = client.GetGrain<IHelloWorld>(0)
+    // Generate random grain number key
+    let key = (Random().Next() % 100)
```

Comment:
Do we need a random key? Not that it's wrong, just curious what you were testing.

Reply:
I made this change to be consistent with: https://github.com/PiotrJustyna/road-to-orleans/blob/003ec9a5b13d01fd4004b088620da41f892f97da/5/Client/src/HelloWorldClientHostedService.cs#L23